src/scripts/main.js

```diff
@@ -1 +1,73 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const handler = () => {
+    resolve();
+    document.addEventListener('click', resolve);
+  };
+
+  document.addEventListener('click', handler);
+
+  setTimeout(reject, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  const handler = () => {
+    resolve();
+    document.addEventListener('click', resolve);
```

Comment:
Same issue here as in line 6. You are adding an event listener inside the handler for the click event. This could potentially lead to a memory leak. Consider removing the event listener once it has been triggered.